test/hotspot/gtest/logging/test_asynclog.cpp

```diff
@@ -253,26 +253,15 @@ TEST_VM_F(AsyncLogTest, droppingMessage) {
   EXPECT_TRUE(file_contains_substring(TestLogFileName, "messages dropped due to async logging"));
 }
 
-TEST_VM_F(AsyncLogTest, StallingModePreventsDroppedMessages) {
-  if (AsyncLogWriter::instance() == nullptr) {
-    return;
-  }
-  set_log_config(TestLogFileName, "logging=debug");
-  LogConfiguration::AsyncMode prev_mode = LogConfiguration::async_mode();
-  LogConfiguration::set_async_mode(LogConfiguration::AsyncMode::Off);
-  test_asynclog_drop_messages();
-  EXPECT_FALSE(file_contains_substring(TestLogFileName, "messages dropped due to async logging"));
-  LogConfiguration::set_async_mode(prev_mode);
-}
-
 TEST_VM_F(AsyncLogTest, stdoutOutput) {
   testing::internal::CaptureStdout();
 
   if (!set_log_config("stdout", "logging=debug")) {
     return;
   }
 
-  bool async = AsyncLogWriter::instance() != nullptr;
+  bool async = AsyncLogWriter::instance() != nullptr
+               && LogConfiguration::async_mode() == LogConfiguration::AsyncMode::Drop;
   if (async) {
```

Comment:
These tests do not currently fail, as the only mode gtests are run in are `drop` mode. However, let's fix them now in order to avoid failures in the future.